src/Pelagos/Bundle/AppBundle/Rabbit/Consumer/DoiConsumer.php

```diff
@@ -286,40 +285,55 @@ protected function deleteDoi($doi, array $loggingContext)
     /**
      * Validate publish for DOI for the dataset.
      *
-     * @param string $restriction    The restriction for the dataset.
-     * @param string $doiStatus      The status of the DOI for the dataset.
-     * @param array  $loggingContext The logging context to use when logging.
+     * @param Dataset $dataset        The dataset.
+     * @param string  $doiStatus      The status of the DOI for the dataset.
+     * @param array   $loggingContext The logging context to use when logging.
      *
      * @return boolean
      */
-    private function validatePublish($restriction, $doiStatus, array $loggingContext)
+    private function validatePublish($dataset, $doiStatus, array $loggingContext)
```

Comment:
I just put that function, so that the cyclomatic complexity of the entire publishDoi() funciton is lower and it is easier for us to read in the future. But i do get your point, it is just used once.